src/main/resources/db/migration/V1_88__eosr_unique_constraint.sql

```diff
@@ -0,0 +1,2 @@
+alter table end_of_service_report
+    add unique (referral_id);
```

Comment:
_danger_
`ALTER TABLE ... ADD CONSTRAINT ... UNIQUE (...)` requires an `ACCESS EXCLUSIVE` lock -- this would block all requests trying to create or read an end of service report

> [`ACCESS EXCLUSIVE`] mode guarantees that the holder is the only transaction accessing the table in any way

Could this be split into a `CONCURRENTLY` unique index and `add constraint {...} unique using index {name}`, please?

Reply:
Well, that is -- if there's any chance of this operation taking longer than 5 seconds